src/api/Shrooms.Infrastructure/Email/MailingService.cs

```diff
@@ -1,47 +1,28 @@
 ﻿using System.Collections.Generic;
 using System.Linq;
-using System.Net.Configuration;
 using System.Net.Mail;
 using System.Threading.Tasks;
-using System.Web;
-using System.Web.Configuration;
 using Microsoft.ApplicationInsights;
 using Microsoft.ApplicationInsights.DataContracts;
 using Microsoft.AspNet.Identity;
 using Shrooms.Contracts.DataTransferObjects;
+using Shrooms.Contracts.Enums;
+using Shrooms.Contracts.Infrastructure;
 using Shrooms.Contracts.Infrastructure.Email;
 
 namespace Shrooms.Infrastructure.Email
 {
     public class MailingService : IMailingService, IIdentityMessageService
     {
+        private readonly EmailBuildingStrategy _emailBuildingStrategy;
+        private readonly ISmtpService _smtpService;
         private readonly TelemetryClient _telemetryClient;
 
-        public MailingService()
+        public MailingService(ISmtpService smtpService, IApplicationSettings appSettings)
```

Comment:
Injecting `ISmtpService` makes MailingService also Smtp, while it could accept some other abstraction, like some service which would send email using SendGrid API. Perhaps `ISmptService` could be something like `IMailSenderService` and `SmtpService` as one of its' implementation.
Yes, in this case there would be a bit too much abstractions over mail sending (IMailingService, IMailSenderService), but it would be at least more extensible. Also it could be an argument, that `IMailinService` is application-related service, while `IMailSenderService` is infrastructure-related service.

Reply:
That's a good point. I've forgot about the architectural plan during implementation.. One of the main reasons of abstracting this into a service is to be able to plug in other notification services like SendGrid or even some SMS service like Twilio :)
I'll rename the interface and it's methods to better suit the abstract role it is meant for.